packages/bull-vault/test/fuzz-test/LeverageBullFuzzTest.t.sol

```diff
@@ -0,0 +1,244 @@
+        vm.deal(auction, _wethToDeposit);
+
+        vm.startPrank(auction);
+        // deposit first WETH in euler
+        IWETH9(weth).deposit{value: _wethToDeposit}();
+        IWETH9(weth).approve(address(leverageBull), _wethToDeposit);
+        leverageBull.depositAndBorrowFromLeverage(_wethToDeposit, 0);
+        vm.stopPrank();
+
+        vm.prank(address(leverageBull));
+        IEulerMarkets(eulerMarketsModule).enterMarket(0, weth);
+
+        uint256 balanceOfUnderlying =
+            IEulerEToken(eToken).balanceOfUnderlying(address(leverageBull));
+        uint256 wethBalanceBefore = IERC20(weth).balanceOf(auction);
+        uint256 balanceOfUnderlyingToWithdraw = balanceOfUnderlying.mul(_percentage).div(100);
+
+        // balanceOfUnderlying is rounded down after deposit, 2wei delta sometimes
+        assertApproxEqAbs(balanceOfUnderlying, _wethToDeposit, 2);
+
+        vm.prank(auction);
+        leverageBull.auctionRepayAndWithdrawFromLeverage(0, balanceOfUnderlyingToWithdraw);
+
+        uint256 wethBalanceAfter = IERC20(weth).balanceOf(auction);
+
+        assertEq(wethBalanceAfter.sub(wethBalanceBefore), balanceOfUnderlyingToWithdraw);
+    }
+
+    function testFuzzingBorrowAndRepayFullBalanceOf(uint256 _wethToDeposit) public {
+        _wethToDeposit = bound(_wethToDeposit, 1e18, 100000000e18);
+
+        vm.deal(auction, _wethToDeposit);
+
+        vm.startPrank(auction);
+        // deposit first WETH in euler
+        IWETH9(weth).deposit{value: _wethToDeposit}();
+        IWETH9(weth).approve(address(leverageBull), _wethToDeposit);
+        leverageBull.depositAndBorrowFromLeverage(_wethToDeposit, 0);
+        vm.stopPrank();
+
+        vm.prank(address(leverageBull));
+        IEulerMarkets(eulerMarketsModule).enterMarket(0, weth);
+
+        uint256 balanceOfUnderlying =
+            IEulerEToken(eToken).balanceOfUnderlying(address(leverageBull));
+
+        uint256 ethUsdPrice = UniOracle._getTwap(ethUsdcPool, weth, usdc, TWAP, false);
+        uint256 usdcToBorrow = _wethToDeposit.wmul(ethUsdPrice).wdiv(TARGET_CR).div(WETH_DECIMALS_DIFF);
+        uint256 eulerUsdcBalance = IERC20(usdc).balanceOf(euler);
+
+        if (usdcToBorrow > eulerUsdcBalance) usdcToBorrow = eulerUsdcBalance;
+
+        uint256 usdcBalanceBefore = IERC20(usdc).balanceOf(auction);
+
+        vm.startPrank(auction);
+        leverageBull.depositAndBorrowFromLeverage(0, usdcToBorrow);
+        vm.stopPrank();
+        
+        uint256 usdcBalanceAfter = IERC20(usdc).balanceOf(auction);
+
+        assertEq(usdcBalanceAfter.sub(usdcBalanceBefore), usdcToBorrow);
+
+        vm.startPrank(auction);
+        uint256 usdcToRepay = IEulerDToken(dToken).balanceOf(auction);
+        IERC20(usdc).approve(address(leverageBull), usdcToRepay);
+        leverageBull.auctionRepayAndWithdrawFromLeverage(usdcToRepay, 0);
+
+        assertEq(IEulerDToken(dToken).balanceOf(auction), 0);
+        assertEq(usdcBalanceAfter.sub(IERC20(usdc).balanceOf(auction)), 0);
```

Comment:
i dont get this assertion - aren't we asserting that the usdc balance doesn't change when we repay?

Reply:
how is this passing?

Reply:
turns out usdcToRepay is getting set to 0, so this is passing. In every test we are repaying 0 usdc